lib/retry.go

```diff
@@ -0,0 +1,20 @@
+package lib
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+func WithRetries(ctx context.Context, retryFn func() error, options struct {
+	retriesCount    int
+	retriesInterval int
+}) error {
```

Comment:
would be nice to allow `options` being nil and set some default values to `retriesCount` and `retriesInterval`

Reply:
I'm thinking if this rule wouldn't bring a complexity that isn't valid now